### PR TITLE
tap gestures for annotation

### LINKED ISSUE #80: Create tap gesture for annotation
- [x] Should de-select other annotation
- [x] Tapping on an annotation toolbar should not create an annotation

### PR DESCRIPTION
**Pull request by darren2pro**
Closes #80

### CODE REVIEW

File: Annotato/Annotato/ViewModels/Document/DocumentViewModel.swift [hunk 1]
```diff
@@ -15,10 +16,32 @@ class DocumentViewModel: ObservableObject {
...
+    }
+
+    func setAllAnnotationsOutOfFocus() {
+        for annotation in annotations {
+            annotation.outOfFocus()
+        }
+    }
+
+    private func setAllOtherAnnotationsOutOfFocus(except annotationInFocus: AnnotationViewModel) {
+        for annotation in annotations where annotation.model.id != annotationInFocus.model.id {
```
Comment (zognin) [inline Annotato/Annotato/ViewModels/Document/DocumentViewModel.swift]:
```suggestion
        for annotation in annotations where annotation.id != annotationInFocus.id {
```

Could we have a computed property in `AnnotationViewModel`?
```
var id: UUID {
    model.id
}
```

File: Annotato/Annotato/ViewModels/Document/DocumentViewModel.swift [hunk 2]
```diff
@@ -15,10 +16,32 @@ class DocumentViewModel: ObservableObject {
...
             return nil
         }
-
         self.model = model
         self.pdfDocument = PdfViewModel(baseFileUrl: baseFileUrl)
         self.annotations = model.annotations.map { AnnotationViewModel(model: $0, document: self) }
+        setUpSubscribersForAllAnnotations()
+    }
+
+    private func setUpSubscribersForAllAnnotations() {
```
Comment (zognin) [inline Annotato/Annotato/ViewModels/Document/DocumentViewModel.swift]:
```suggestion
    private func setUpSubscribers() {
```

When individual annotations are added, we set up subscribers for them separately

**Review by zognin — COMMENTED** [review]
LGTM just 2 small suggestions

File: Annotato/Annotato/ViewModels/Document/DocumentViewModel.swift (hunk 1 above)
Comment (darren2pro) [reply Annotato/Annotato/ViewModels/Document/DocumentViewModel.swift]: Ok done

File: Annotato/Annotato/ViewModels/Document/DocumentViewModel.swift (hunk 2 above)
Comment (darren2pro) [reply Annotato/Annotato/ViewModels/Document/DocumentViewModel.swift]: Ok done too

**Review by zognin — APPROVED** [review]
LGTM thanks!

**Pull request merged**